kubectl-cloudflow/cloudflowavro/avrocompatibility.go

```diff
@@ -0,0 +1,48 @@
+package cloudflowavro
+
+import (
+	"github.com/hamba/avro"
+	"github.com/linkedin/goavro/v2"
+)
+
+func ValidateSchema(schema string) (*string, error) {
+	codec, err := goavro.NewCodec(schema)
+	if err != nil {
+		return nil, err
+	} else {
+		canonical := codec.CanonicalSchema()
+		return &canonical, nil
+	}
+}
+
+type ReaderSchemaValidationError struct {
+	msg string
+}
+
+type WriterSchemaValidationError struct {
+	msg string
+}
+
+func (e *ReaderSchemaValidationError) Error() string { return e.msg }
+
+func (e *WriterSchemaValidationError) Error() string { return e.msg }
+
+func CheckSchemaCompatibility(reader string, writer string) error {
+    _, err := ValidateSchema(reader)
+
+	if err != nil {
+		return &ReaderSchemaValidationError{msg: err.Error()}
+	}
+
+	_, err = ValidateSchema(writer)
+
+	if err != nil {
+		return &WriterSchemaValidationError{msg: err.Error()}
+	}
+
+	readerSchema, err := avro.ParseWithCache(reader, "", avro.DefaultSchemaCache)
+	writerSchema, err := avro.ParseWithCache(writer, "", avro.DefaultSchemaCache)
+	s := avro.NewSchemaCompatibility()
```

Comment:
Can we implement it so you don't have to new an instance? is this for performance reasons? have you seen how much difference it makes to use a cache, or use no cache?

Reply:
I guess it is nice for verifying many connections, which might use the same schemas.. although you could argue that you could deduce from similar connections that you only need to verify every combination once..

Reply:
A new connection just cleans up the cache. I dont think this would be a problem for normal use unless you make million connections.